lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3432,7 +3432,9 @@ bool SPIRVToLLVM::translate() {
   if (!transSourceExtension())
     return false;
   transGeneratorMD();
-  if (!lowerBuiltins(BM, M))
+  SPIRVWord Ver = 0;
+  SourceLanguage SrcLang = BM->getSourceLanguage(&Ver);
```

Comment:
We should not be using a SPIR-V module's source language to affect the translation flow.  The SPIR-V specification for `OpSource` states:

> This has no semantic impact and can safely be removed from a module.

Reply:
Hmm. But name mangling happens in a different way depending on what the source language is. If we have to unmangle the names. we need to know the source language. May be we can do something in the regularization pass?

Thanks